File: java/15_enums.java
public class Enums {
    enum Status {
        PENDING, RUNNING, DONE, FAILED
    }

    public static void main(String[] args) {
        Status status = Status.RUNNING;
        System.out.println(status);

        if (status == Status.RUNNING) {
            System.out.println("still working...");
        }

        for (Status s : Status.values()) {
            System.out.println(s);
        }
    }
}

File: typescript/15_enums.ts
// 15 enums

enum Status {
  Pending = "pending",
  Running = "running",
  Done = "done",
  Failed = "failed",
}

const status: Status = Status.Running;
console.log(status);

if (status === Status.Running) {
  console.log("still working...");
}
